interfaces/builtin/fwupd.go

```diff
@@ -246,6 +256,41 @@ func (iface *fwupdInterface) AppArmorConnectedPlug(spec *apparmor.Specification,
 	}
 	snippet := strings.Replace(fwupdConnectedPlugAppArmor, old, new, -1)
 	spec.AddSnippet(snippet)
+
+	if release.OnClassic {
+		// We only need the mount rules on core systems
+		return nil
+	}
+
+	// Allow mounting boot partition to snap-update-ns
+	emit := spec.AddUpdateNSf
+	for _, target := range bootPartitionDirs {
+		source := "/var/lib/snapd/hostfs" + target
+		emit("  # Read-write access to %s\n", target)
+		emit("  mount options=(bind) %s/ -> %s/,\n", source, target)
+		emit("  remount options=(bind, rw) %s/,\n", target)
+		emit("  umount %s/,\n\n", target)
+	}
+	return nil
+}
+
+func (iface *fwupdInterface) MountConnectedPlug(spec *mount.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
```

Comment:
As with the AppArmor rules, this should be `MountPermanentSlot`.  Same suggestion about using `implicitSystemPermanentSlot(slot)` in place of `release.OnClassic`.